### DISCUSSION TITLE
JS client sort parent table by foreign key, throwing error

### DISCUSSION THREAD

**Discussion by orenlande** [supabase/supabase#16655] [Questions]
Hello,

I have the following tables:
```
decks
| deck_id   | ... | ... |
| 1         | ... | ... |

deck_views
| deck_id --> foreign key decks(deck_id) | views_count | ... | ... |
| 1                                      | 1000        | ... | ... |

constraint deck_views_deck_id_fkey foreign key (deck_id) references decks (deck_id) on delete cascade
```

when running the following query, I get the next error:
```
await supabase
      .from("decks")
      .select(
        "*, deck_views(views_count)"
      ).order("deck_views(views_count)", {
        ascending: false,
      });

{
    "code": "PGRST118",
    "details": "'decks' and 'deck_views' do not form a many-to-one or one-to-one relationship",
    "hint": null,
    "message": "A related order on 'deck_views' is not possible"
}
```

I have been looking online for resources, and seems like the mentioned way is the only possible way to sort parent table with foreign keys. Not sure what I've done wrong. 

Thanks for any help in advance!

**Comment by GaryAustin1** [comment]
Basically the .order() with embedded only works if your embedded table foreign key is unique.  It is the "one" side of the error message.
![image](https://github.com/supabase/supabase/assets/54564956/52859fc4-da34-4c6e-bb48-0aa45fac1f6a)
So if you embedded table can have multiple values for a parent table row then it won't work.
This may also be restricted that the foreign key has to be in decks and point to deck_views, but that I do not know.

**Reply by orenlande** [reply]
Sorry, I'm not sure I understood your explanation. Here is a visualization of the relevant tables. Maybe it will explain my situation better. 

If I understand correctly, do you suggest I should add a column in `decks` that refers to the table `deck_views`?

![2CF56E16-A6BE-4EAD-A695-E080A4D4BA91](https://github.com/supabase/supabase/assets/20873769/752479f8-90d8-43a5-858c-e17c17efc0ce)
